modules/core/shared/src/main/scala/io/circe/Printer.scala

```diff
@@ -334,4 +334,47 @@ final object Printer {
       }
     }
   }
+
+  private[circe] final class AppendableByteBuffer(cs: Charset) extends Appendable {
+    private[this] var index = 0
+    private[this] var chars = new Array[Char](32)
+
+    private[this] def ensureToFit(n: Int): Array[Char] = {
```

Comment:
No particular reason! I've thnk I've seen this pattern somewhere recently and decided to try it out. Let me change that to a `Unit` function so we always mutate `chars`.